core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java

```diff
@@ -110,6 +108,10 @@ public InnerHitsContext innerHits() {
         public void setChildInnerHits(Map<String, InnerHitsContext.BaseInnerHits> childInnerHits) {
             this.childInnerHits = new InnerHitsContext(childInnerHits);
         }
+
+        boolean doCountOnly() {
+            return from() < 0 || from() >= size() || size() <= 0;
+        }
```

Comment:
@jpountz The reason I took this approach, was that `TopDocsCollector#topDocs(...)` returns a top docs with total hits set to 0 if `from` is larger than number of collected docs, which is unexpected. Instead it should just return a TopDocs with no docs but a real totalHits/maxScore?

In the search api we don't rely on `TopDocsCollector#topDocs(...)`, but rather on `TopDocs#merge`, which always set maxScore and totalHits.